internal/services/storage/storage_account_resource.go

```diff
@@ -1991,7 +1991,13 @@ func resourceStorageAccountUpdate(d *pluginsdk.ResourceData, meta interface{}) e
 			return fmt.Errorf("unable to locate %s", *id)
 		}
 
-		queueClient, err := storageClient.QueuesDataPlaneClient(ctx, *account, storageClient.DataPlaneOperationSupportingAnyAuthMethod())
+		// check authentication method.
+		authMethod := storageClient.DataPlaneOperationSupportingAnyAuthMethod()
+		if !d.Get("shared_access_key_enabled").(bool) || d.Get("network_rules") != nil {
```

Comment:
It's not clear why we would change the possible authentication methods based on whether there are any network rules on the storage account?

If there is a link I'm not understanding - then how would this determine whether those rules do or don't include the Terraform CLI host?